### ISSUE TITLE
[Bug] More and more things don't load

### ISSUE THREAD

**Issue by Geartwo** [simibubi/Above-and-Beyond#128]
I think the modpack have big NBT problem.
World runs on a server started with V1.0, no extra mod added or removed.

Create Stuff often only load when after you look at it.
![2021-11-14_15 47 54](https://user-images.githubusercontent.com/4224642/141686839-1e02cf5b-db8e-4603-bf0f-8c85255734fc.png)
![2021-11-14_15 49 03](https://user-images.githubusercontent.com/4224642/141686887-ce70889b-bbbb-4be9-a638-f048948a9144.png)

Drawers are shown as empty until you put the same item in it again
![2021-11-14_15 48 24](https://user-images.githubusercontent.com/4224642/141686806-9401c2e6-7074-45d6-8c1c-8c2786230480.png)

Signs don't load and when they load they are empty and heads only show after you look at them
![2021-11-14_15 48 33](https://user-images.githubusercontent.com/4224642/141686949-634ffd20-5668-4d22-806a-0189ff4640f4.png)
![2021-11-14_15 48 36](https://user-images.githubusercontent.com/4224642/141686989-16818dc5-329c-4b3a-a8e5-2bca67a15eb0.png)

Do someone have the same symptoms?

**Comment by BD103** [comment]
This is probably a visual/client issue. You can try resetting video settings. Or if you have Optifine, you may want to remove it.

**Comment by Johnystar** [comment]
> This is probably a visual/client issue. You can try resetting video settings. Or if you have Optifine, you may want to remove it.

Nope, this is happening to me as well and it is a client/server desync. Sometimes, drawers appear empty on rejoining, I try inserting a random into them, my hand slot even appears empty, but then I try clicking the slot in my inventory and the items are still there, because, on the server's side, there already is other stuff in the drawer, the client just didn't seem to get that information for whatever reason. (this information gets synced when I shift + right-click the drawer with an empty hand, which opens the drawer's menu)

This is a pretty big problem. So far I've seen this happen with:

- **beds** (appear before me again when I look at them)
- **Create belts** (some parts don't appear to be moving sometimes on the client-side, even though items on them do still move just fine)
- **drawers** (drawers often appear empty or not locked (even though they are) on the client-side)
- **flower boxes** (appear empty, right-clicking re-syncs)
- **signs** (occasionally appear empty)
- _probably other stuff_

I'm playing on a Create: Above and Beyond v1.1 server with a Create: Above and Beyond v1.1 client - no mod customisations, additions or anything similar have been made.

**Comment by DAmNRelentless** [comment]
I only had this happening once and some belts were invisible. It was fixed by just relogging.
I didn't have such a large amount of stuff being invisible yet so this might be a sychronization issue in general.

**Comment by Geartwo** [comment]
The best workaround i have found so far is to change dimension (Nether and back).
Then at least all chests and drawers are send to the client and are usable.

**Comment by Conner-Anderson** [comment]
I also have this problem, and no graphical adjustment that I have tried has had any effect. The issue is constant when I am on the server alone, but sometimes fixes itself completely when other players are online.
For the record, I've tried: uninstalling optifine, disabling Flywheel, adjusting every graphical setting vaguely related to entity rendering. This confirms for me that it is not a graphical bug, as others have said.

**Are any of you folks playing on the same network the server is running on? My server runs on the same machine.
Restarting the server fixes errors right around where I spawn, but not the rest.**

In some areas, the problem happens with the same entities over-and-over, but not neighboring ones. Four chests on the same y-level on a chest wall will disappear, while the rest remain.

Additional items it happens with past Johnystar's list:
- Chests
- Shafts
- Gears (both kinds)
- Hanging signs (invisible, then blank when they appear)

None of the four other people on the server have these issues. I am using MultiMC to launch, but found the same issue with ATLauncher and CurseForge.

**Comment by Conner-Anderson** [comment]
More things I tried:

- Updated AdvancedRocketry and LibVulpes to the latest versions. No Change.
- Loaded same world in singleplayer. Fixed all issues.
- Connected to server from another IP. No Change.

**Comment by Conner-Anderson** [comment]
Solution: Do not use microblocks.

The science:

1. Placed drawers with items, chests, and written signs in a distant chunk in creative.
2. Relogged. Observed no errors.
3. Scattered microblocks (from CBMultipart) around the area
4. Relogged. Observed invisible item labels on drawers, invisible signs. Chests remained.
5. Removed microblocks from area.
6. Relogged. Observed no errors.
7. Added a ProjectRed component to the area.
8. Relogged. Observed no errors.
9. Added a redstone wire to the same block as the ProjectRed component.
10. Relogged. Observed invisibility on _some_ of the drawer labels.

This shows that the presence of microblocks in the area (perhaps the chunk?) will cause rendering (or sync?) errors with chests, signs, drawers, and other tile entities. ProjectRed components are safe as long as one doesn't use their micro-block features. 

Terribly disappointing. I tried updating CBMultipart (and CodeChickenLib) to their latest versions with no result.

**Comment by Conner-Anderson** [comment]
The plot thickens:
A server with only CBMultipart (the microblocks mod) and its dependencies does not develop the issue on a fresh world as hard as I try. It does, though, have the issue on a world that formerly ran the full modpack. So there's something that the modpack does to a world that makes CBMultipart cause bugs. Yes, I tested that same world with no mods apart from forge, and the effect did not remain. It seems that the issue arises in an area when microblocks are nearby, and on a world that formerly ran the full modpack.

I could try to narrow down what mod causes the issue further, but that's gonna take a long while...

**Comment by Dalarrus** [comment]
I have this exact same issue(Mostly as Chests and Signs being invisible/losing text, as well as drawers.) and went through the same rigors of disabling Optifine/Performant/Flywheel and having the issue persist, I am also connecting across a local network to the server, often as the only/first person, as someone had mentioned that being a possible contributing factor.

As was helpfully mentioned, removing all the C&B Microblocks around the area resolves the issue entirely.

**Comment by Conner-Anderson** [comment]
**Improved Solution!**
By using the [forceload](https://minecraft.fandom.com/wiki/Commands/forceload) operator command to keep a few nearby chunks loaded, the issue can be mitigated. Only a single chunk need be loaded within 10-16 chunks (don't know yet) of the problem area to help. When a chunk is loaded like this, the problem either goes away or goes away on relog, unlike before.

On my server, I've just force-loaded a chunk each of my friends' bases and the community area. Now whenever I'm the first to join the server, I disconnect and reconnect to solve the issue even if no one else comes online.

This should have minimal performance impact, since these isolated chunks won't receive random block ticks for plant growth.

Why does this happen? Space magic, duh.
